lib/vty.c

```diff
@@ -2227,6 +2250,7 @@ static int vtysh_flush(struct vty *vty)
 		vty_close(vty);
 		return -1;
 	case BUFFER_EMPTY:
+		vty->vty_buf_size_written = 0;
```

Comment:
so the counter is only reset if obuf is empty?  if the write partially succeeds, we'll stay in a "flush" loop with every vty_out() call - is that ... what you want?

Reply:
Yes, there was an extra 2-3 MB used with reset when it hits the 128 KB and we will still have our 25-30 % memory reduction, so I changed it and it will reset in both the places and we won't be in flush loop for every vty_out call and it calls only after we reach the chunk is 128KB again